packages/flutter/test/rendering/paragraph_test.dart

```diff
@@ -911,6 +912,392 @@ void main() {
       expect(geometry2.hasContent, true);
       expect(geometry2.status, SelectionStatus.uncollapsed);
     });
+
+    test('can granularly extend selection - character', () async {
+      final TestSelectionRegistrar registrar = TestSelectionRegistrar();
+      final List<RenderBox> renderBoxes = <RenderBox>[];
+      final RenderParagraph paragraph = RenderParagraph(
+        const TextSpan(
+            children: <InlineSpan>[
+              TextSpan(text: 'how are you\nI am fine\nThank you'),
+            ]
+        ),
+        textDirection: TextDirection.ltr,
+        registrar: registrar,
+        children: renderBoxes,
+      );
+      layout(paragraph);
+
+      expect(registrar.selectables.length, 1);
+      selectionParagraph(paragraph, const TextPosition(offset: 4), const TextPosition(offset: 5));
+      expect(paragraph.selections.length, 1);
+      TextSelection selection = paragraph.selections[0];
+      expect(selection.start, 4); // how `a`re you
+      expect(selection.end, 5);
+
+      // Equivalent to sending shift + arrow-right
+      registrar.selectables[0].dispatchSelectionEvent(
+        const GranularlyExtendSelectionEvent(
+          forward: true,
+          isEnd: true,
+          granularity: TextGranularity.character,
+        ),
+      );
+      selection = paragraph.selections[0];
+      expect(selection.start, 4); // how `ar`e you
```

Comment:
I think elsewhere in our tests I have seen `how [ar]e you` or `how |ar|e you`.  Maybe we should standardize at some point.

No action needed though, just thinking out loud.